src/librustc_mir/interpret/machine.rs

```diff
@@ -254,6 +254,14 @@ pub trait Machine<'mir, 'tcx>: Sized {
         kind: Option<MemoryKind<Self::MemoryKind>>,
     ) -> (Cow<'b, Allocation<Self::PointerTag, Self::AllocExtra>>, Self::PointerTag);
 
+    /// Called to notify the machine of deallocations
+    fn notify_deallocation(
```

Comment:
To keep with the style of other comparable hooks, please call this `before_dealloc` or `before_deallocation`.